osmdroid-shape/src/main/java/org/osmdroid/shape/DefaultShapeMetaSetter.java

```diff
@@ -0,0 +1,74 @@
+package org.osmdroid.shape;
+
+import android.graphics.Paint;
+
+import net.iryndin.jdbf.core.DbfRecord;
+
+import org.osmdroid.views.overlay.Marker;
+import org.osmdroid.views.overlay.Overlay;
+import org.osmdroid.views.overlay.PolyOverlayWithIW;
+import org.osmdroid.views.overlay.Polygon;
+
+import java.nio.charset.Charset;
+import java.text.ParseException;
+
+public class DefaultShapeMetaSetter implements ShapeMetaSetter {
+
+    private String metaString;
+    private String subString;
+    private Overlay mOverlay;
+    private String shapeType;
+
+    @Override
+    public void set(DbfRecord metadata, Overlay overlay) throws ParseException {
+        if (metadata != null){
+            metadata.setStringCharset(Charset.defaultCharset());
+            metaString = metadata.toMap().toString();
+        } else{
+            metaString = "";
+        }
+
+        shapeType = overlay.getClass().getName();
+
+        if (shapeType.equals("org.osmdroid.views.overlay.Marker"))
+            subString = ((Marker) overlay).getPosition().toString();
+        else if (shapeType.equals("org.osmdroid.views.overlay.Polygon") || shapeType.equals("org.osmdroid.views.overlay.Polyline"))
+            subString = overlay.getBounds().toString();
+        else
+            subString = "";
+
+        mOverlay = overlay;
+    }
+
+    @Override
+    public String getSnippet() {
+        return metaString;
+    }
+
+    @Override
+    public String getTitle() {
+        if (metaString.length() > 100) {
+            return metaString.substring(0, 96) + "...";
+        }
+        return metaString;
+    }
+
+    @Override
+    public String getSubDescription() {
+        return subString;
+    }
+
+    @Override
+    public Paint getFillPaint() {
+        if (shapeType.equals("org.osmdroid.views.overlay.Polygon"))
+            return ((Polygon) mOverlay).getFillPaint();
+        return null;
+    }
+
+    @Override
```

Comment:
we don't need that.

Reply:
we don't need that.

Oh, I understand the confusion: the point is not to retrieve the Overlay fill paint or stroke paint, the point is to set it.
Basically: standard case, I don't really care about the colors, I'm OK with the default values of Polygon and Polyline painting. It's cool: I do nothing, I don't override the defaults.
Refined case: I want the polygons to have a red stroke color and the polylines a blue one. For Polygons it's going to be written `polygon.getOutlinePaint().setColor(Color.RED)`. And something similar for polylines.

The point is: the code inside `ShapeConverter` doesn't care at all about what you want to change for your polygons, color, title, snippet, ..., because the `ShapeMetaSetter` will do everything about it (first step: computing the values, second step: setting the values for the polygon).